posthog/models/feature_flag/flag_matching.py

```diff
@@ -1,265 +1,29 @@
 import hashlib
-import json
 from dataclasses import dataclass
 from enum import Enum
-from typing import Any, Dict, List, Optional, Tuple, Union, cast
+from typing import Any, Dict, List, Optional, Tuple, Union
 
-from django.core.cache import cache
-from django.db import DatabaseError, models
+from django.db import DatabaseError
 from django.db.models.expressions import ExpressionWrapper, RawSQL
 from django.db.models.fields import BooleanField
 from django.db.models.query import QuerySet
-from django.db.models.signals import pre_delete
-from django.utils import timezone
-from rest_framework.exceptions import ValidationError
 from sentry_sdk.api import capture_exception
 
-from posthog.client import sync_execute
-from posthog.constants import AvailableFeature, PropertyOperatorType
-from posthog.models.cohort import Cohort
-from posthog.models.experiment import Experiment
 from posthog.models.filters import Filter
 from posthog.models.filters.mixins.utils import cached_property
 from posthog.models.group import Group
 from posthog.models.group_type_mapping import GroupTypeMapping
-from posthog.models.organization import OrganizationMembership
 from posthog.models.person import Person, PersonDistinctId
 from posthog.models.property import GroupTypeIndex, GroupTypeName
-from posthog.models.property.property import Property, PropertyGroup
-from posthog.models.signals import mutable_receiver
-from posthog.models.team.team import Team
+from posthog.models.property.property import Property
 from posthog.queries.base import match_property, properties_to_Q
 
-
-class FeatureFlag(models.Model):
```

Comment:
Kind of a nit but it'd be nice if the feature flag model file stayed as the feature flag model file and the flag matching code was pulled out to a new file vs turning the feature flag model file into the flag matching file if that makes sense? 

I think this way it can keep some git history that would be easy to see, especially when coding? For the feature flag model file 😅

Reply:
it's out of my control :P - that's what github decided happened, based on most lines of code moving to which file

Reply:
ohhh that sucks 😆 whatever then!

Reply:
fwiw I sort-of prefer it like this, since we often change the matching code, but models haven't changed since the beginning, so you'd want the diff on matching